Core.ApplicationServices/UserService.cs

```diff
@@ -344,5 +371,26 @@ private static void Delete(User user)
             user.HasApiAccess = false;
             user.HasStakeHolderAccess = false;
         }
+
+        private Result<UserDeletionStrategy, OperationError> GetUserDeletionStrategy(int orgId, User user)
+        {
+            if (user.Deleted)
+                return new OperationError(OperationFailure.BadInput);
+            if (user.OrganizationRights.Any(x => x.OrganizationId == orgId) == false)
```

Comment:
use the convenience method `GetOrganizationIds` on User to check the ids in stead